Sources/IBLinterKit/Rules/ColorResourcesRule.swift

```diff
@@ -0,0 +1,71 @@
+import Foundation
+import IBDecodable
+
+extension Rules {
+
+    struct ColorResourcesRule: Rule {
+
+        static let identifier = "color_resources"
+        static let description = "Check if named color resources are valid."
+
+        let assetsCatalogs: [AssetsCatalog]
+
+        public init(context: Context) {
+            let paths = glob(pattern: context.workDirectory.appendingPathComponent("**/*.xcassets").path)
+            let excluded = context.config.excluded.flatMap { glob(pattern: "\($0)/**/*.xcassets") }
+            let lintablePaths = paths.filter { !excluded.map { $0.absoluteString }.contains($0.absoluteString) }
+            self.init(catalogs: lintablePaths.map { AssetsCatalog.init(path: $0.relativePath) })
+        }
+
+        init(catalogs: [AssetsCatalog]) {
+            self.assetsCatalogs = catalogs
+        }
+
+        func validate(xib: XibFile) -> [Violation] {
+            return validate(
+                for: xib.document.children(of: NamedColor.self),
+                file: xib
+            )
+        }
+
+        func validate(storyboard: StoryboardFile) -> [Violation] {
+            return validate(
+                for: storyboard.document.children(of: NamedColor.self),
+                file: storyboard
+            )
+        }
+
+        private func validate<T: InterfaceBuilderFile>(for colors: [NamedColor], file: T) -> [Violation] {
+            let catalogAssetNames = assetsCatalogs.flatMap { $0.values }
+            return colors
+                .map { $0.name }
+                .filter { !catalogAssetNames.contains($0) }
+                .map {
+                    Violation(
+                        pathString: file.pathString,
+                        message: "\($0) not found",
+                        level: .error)
+            }
+        }
+    }
+}
+
+private extension AssetsCatalog {
+    // namespaced names of the
+    var values: [String] {
+        return entries.flatMap { $0.values }
+    }
+}
+
+private extension AssetsCatalog.Entry {
+    var values: [String] {
+        switch self {
+        case .group(_, let items):
+            return items.flatMap { $0.values }
+        case .color(_, let value):
+            return [value]
+        case .image(_, let value):
+            return [value]
+        }
+    }
+}
```

Comment:
These private extensions are copied from ImageResourceRule.swift.
What do you think about best position of this extensions?

1. keep private extensions in each files.
2. escalate to internal extension and move to inside of AssetsCatalog.swift

Reply:
hmm, xcassets can have same name different assets entry (for example, image and color).
We should filter to only color assets for this rule. (and also filter to image for `image_resources` rule)

okey, keep private extensions in each files is good.

Reply:
okey, fixed for that case. (only for color_resources rule in this PR)

Reply:
so there is a bug in https://github.com/IBDecodable/IBLinter/blob/master/Sources/IBLinterKit/Rules/ImageResourcesRule.swift#L90

maybe if the code is moved to AssetsCatalog, rename it to `colorNames` and `imageNames` to be clear about the purpose of the accessors
so no more copy paste, or adding new case (like `color`) , which could create bugs

Reply:
Should i fix by moving extension methods to AssetsCatalog.swift and rename to `colorNames` and `imageNames` in this PR?

Reply:
Yes, you can

Reply:
@phimage I try to filter by `Item` enum argument having method instead of `xxxNames` methods https://github.com/r-plus/IBLinter/commit/df983e6508109fe0185260a6c45dbe0d914fcc08
What do you think of this strategy? (not yet applied for this PR)

Reply:
it's more "swifty" and maybe more elegant but
this add just a little runtime cost
so as you want

Reply:
I checked how many runtime cost by unit test https://github.com/r-plus/IBLinter/commit/e7d939c3e56ea0daf2dccec41755f6f66062de21

Results in my machine

<img width="368" alt="スクリーンショット 2020-01-29 10 42 31" src="https://user-images.githubusercontent.com/425216/73320869-1a257480-4284-11ea-9876-32c0cf782edb.png">

using enum for filter is about 10% slower than type fixed method. But it is single digit ms level cost in usual size of xcassets. so I want to use swifty way.
is it good?

Reply:
okey, picked that commit.

Reply:
yes, you can use enum